lib/xcodeproj/scheme/environment_variables.rb

```diff
@@ -0,0 +1,117 @@
+        all_variables.each { |existing_var| remove_variable(existing_var) if existing_var.key == env_var.key }
+        @xml_element.add_element(env_var.xml_element)
+        @all_variables << env_var
+      end
+
+      # @param [EnvironmentVariable,Hash{String => String}] variable
+      #   The key value pairs currently set in @xml_element
+      #
+      # @return [Array{EnvironmentVariable}]
+      #   The new set of environment variables after removal
+      #
+      def remove_variable(variable)
+        env_var = variable.is_a?(EnvironmentVariable) ? variable : all_variables.select { |var| var.key == variable[:key] }[0]
+        raise "Unexpected parameter type: #{env_var.class}" unless env_var.is_a?(EnvironmentVariable)
+        @xml_element.delete_element(env_var.xml_element)
+        @all_variables -= [env_var]
+      end
+
+      # @param [EnvironmentVariable,Hash{String => String}] variable
+      #   The key value pairs currently set in @xml_element
+      #
+      def to_a
+        all_variables.map(&:to_h)
+      end
+    end
+
+    # This class wraps the EnvironmentVariable node of a .xcscheme XML file.
+    # Environment variables are accessible via the NSDictionary returned from
+    # [[NSProcessInfo processInfo] environment] in your app code.
+    #
+    class EnvironmentVariable < XMLElementWrapper
+      # @param [REXML::Element,EnvironmentVariable,Hash{String => String}] value
+      #   The 'EnvironmentVariable' XML node that this object represents.
+      #   If nil, it will create a default XML node to use. If a hash, it
+      #   must contain keys 'key' and 'value', and optionally have 'enabled'.
+      #   If not supplied, it will default to being enabled.
+      #
+      def initialize(value)
+        create_xml_element_with_fallback(value, VARIABLE_NODE) do
+          raise "Must pass a Hash with 'key' and 'value'!" unless value.is_a?(Hash) && value.key?(:key) && value.key?(:value)
+
+          @xml_element.attributes['key'] = value[:key]
+          @xml_element.attributes['value'] = value[:value]
+          @xml_element.attributes['isEnabled'] = value.key?(:enabled) ? bool_to_string(value[:enabled]) : bool_to_string(true)
+        end
+      end
+
+      def key
+        @xml_element.attributes['key']
+      end
+
+      def key=(key)
+        @xml_element.attributes['key'] = key
+      end
+
+      def value
+        @xml_element.attributes['value']
+      end
+
+      def value=(value)
+        @xml_element.attributes['value'] = value
+      end
+
+      def enabled
+        string_to_bool(@xml_element.attributes['isEnabled'])
+      end
+
+      def enabled=(enabled)
+        @xml_element.attributes['isEnabled'] = bool_to_string(enabled)
+      end
+
+      # @return [Hash{:key => String, :value => String, :enabled => Boolean}]
+      #   The environment variable XML node with attributes converted to Hash keys
+      def to_h
```

Comment:
Seems to be differing information. From what I've read, appears that `to_hash` is specifically if the object acts like a hash and can be converted with no loss of fidelity. I don't think that is the case here. I don't feel strongly though, so LMK if I'm just misunderstood.